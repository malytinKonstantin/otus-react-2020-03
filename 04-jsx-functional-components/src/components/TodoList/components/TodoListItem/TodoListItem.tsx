import React from 'react'

export interface TodoListItem {
  id: string
  title: string
  isCompleted: boolean
}

interface TodoListItemProps extends TodoListItem {
  onDone: () => void
  onActive: () => void
  onRemove: () => void
}

export const TodoListItem: React.FC<Props> = (props) => {
  const { title, description, isCompleted, onDone, onActive, onRemove } = props

  return (
    <div>
      <div className="title">{title}</div>
      <div className="nav">
        {isCompleted ? (
          <button onClick={onActive}>взять в работу</button>
        ) : (
          <button onClick={onDone}>выполнено</button>
        )}
        <button onClick={onRemove}>удалить</button>
      </div>
    </div>
  )
}
