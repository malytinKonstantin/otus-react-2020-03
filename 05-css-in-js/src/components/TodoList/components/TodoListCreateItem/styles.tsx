import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 15px;

  & * {
    margin-right: 15px;
  }
`
export const TextField = styled.input`
  padding: 5px 10px;
  border-radius: 5px;
  box-shadow: none;
  border: 1px solid #afafaf;
`
export const Button = styled.button`
  border: none;
  color: #fff;
  background: #1d96ff;
  padding: 5px 15px;
  border-radius: 3px;
`

export const ButtonReset = styled.button`
  ${Button} {
  }
`

export const ButtonSubmit = styled.button`
  ${Button} {
  }
`
