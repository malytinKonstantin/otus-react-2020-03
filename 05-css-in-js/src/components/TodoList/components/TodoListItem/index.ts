export * from './TodoListItem'
export * from './styles'
