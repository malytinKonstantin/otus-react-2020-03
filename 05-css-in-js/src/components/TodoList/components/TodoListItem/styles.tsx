import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const Title = styled.div`
  color: #000;
`

const ButtonBase = css`
  &:active {
    outline: none;
    box-shadow: none;
  }
`

export const ButtonToActive = styled.button`
  ${ButtonBase}
  background-color: green;
`

export const ButtonToDone = styled.button`
  ${ButtonBase}
  background-color: yellow;
`
