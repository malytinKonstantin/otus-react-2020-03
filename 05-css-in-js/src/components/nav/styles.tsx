import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const NavWrapper = styled.nav`
  display: flex;
  margin-bottom: 30px;
`

export const NavList = styled.ul`
  display: flex;
  align-items: center;
  list-style-type: none;
  padding: 0;
  margin: 0;

  & > * {
    margin-right: 15px;
  }
`

export const User = styled.div`
  margin-left: auto;
  display: flex;
  align-items: center;
`

export const UserName = styled.div`
  margin-right: 15px;
`
