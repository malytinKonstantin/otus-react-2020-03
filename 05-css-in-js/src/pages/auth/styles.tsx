import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const Row = styled.div`
  margin-bottom: 15px;
`

export const Wrapper = styled.div`
  margin: 0 auto;
  padding-top: 100px;
  width: 200px;
`

export const Label = styled.label`
  display: block;
`

export const Input = styled.input`
  width: 100%;
`
