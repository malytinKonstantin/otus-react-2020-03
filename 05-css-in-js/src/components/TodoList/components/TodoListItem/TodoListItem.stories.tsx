import React from 'react'
import { action } from '@storybook/addon-actions'
import { withKnobs, number, text, boolean } from '@storybook/addon-knobs'
import { TodoListItem } from './TodoListItem'

export default {
  title: 'Todo list item',
  decorators: [withKnobs],
}

export const TodoListItemExample: React.FC<{}> = () => {
  const props = {
    id: number('id', 0),
    title: text('title', 'test text'),
    isCompleted: boolean('boolean', false),
    onDone: action('done task'),
    onActive: action('to active task'),
    onRemove: action('remove task'),
  }

  return <TodoListItem {...props} />
}
