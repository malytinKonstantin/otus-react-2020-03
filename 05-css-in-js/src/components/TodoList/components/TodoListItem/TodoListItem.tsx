import React from 'react'
import { Title, ButtonToActive, ButtonToDone } from './styles'

export interface TodoListItem {
  id: string
  title: string
  isCompleted: boolean
}

interface TodoListItemProps extends TodoListItem {
  onDone: () => void
  onActive: () => void
  onRemove: () => void
}

export const TodoListItem: React.FC<Props> = (props) => {
  const { title, isCompleted, onDone, onActive, onRemove } = props

  return (
    <div>
      <Title>{title}</Title>
      <div className="nav">
        {isCompleted ? (
          <ButtonToActive onClick={onActive}>взять в работу</ButtonToActive>
        ) : (
          <ButtonToDone onClick={onDone}>выполнено</ButtonToDone>
        )}
        <button className="btn-remove" onClick={onRemove}>
          удалить
        </button>
      </div>
    </div>
  )
}
