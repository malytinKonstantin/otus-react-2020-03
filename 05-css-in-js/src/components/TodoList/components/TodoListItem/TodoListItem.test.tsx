import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { Title, ButtonToActive, ButtonToDone } from './styles'
import { TodoListItem } from './TodoListItem'

describe('<TodoListItem />', () => {
  const testItem = {
    id: 123,
    title: 'test-title',
    isCompleted: false,
    onDone: jest.fn,
    onActive: jest.fn,
    onRemove: jest.fn,
  }

  it('renders snapshot', () => {
    expect(
      renderer.create(<TodoListItem {...testItem} />).toJSON(),
    ).toMatchSnapshot()
  })

  it('Default render', () => {
    const wrapper = shallow(<TodoListItem {...testItem} />)
    expect(wrapper.find(Title).text()).toEqual(testItem.title)
    expect(wrapper.find(ButtonToDone)).toHaveLength(1)
    expect(wrapper.find(ButtonToActive)).toHaveLength(0)
  })

  it('Render item with isCompleted: true ', () => {
    testItem.isCompleted = true
    const wrapper = shallow(<TodoListItem {...testItem} />)
    expect(wrapper.find(ButtonToDone)).toHaveLength(0)
    expect(wrapper.find(ButtonToActive)).toHaveLength(1)
  })
})
