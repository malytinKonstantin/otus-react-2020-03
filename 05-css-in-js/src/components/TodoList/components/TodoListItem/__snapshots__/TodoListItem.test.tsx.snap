// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem /> renders snapshot 1`] = `
<div>
  <div
    className="css-k24i6u"
  >
    test-title
  </div>
  <div
    className="nav"
  >
    <button
      className="css-1noqwe3"
      onClick={[Function]}
    >
      выполнено
    </button>
    <button
      className="btn-remove"
      onClick={[Function]}
    >
      удалить
    </button>
  </div>
</div>
`;
